Scripts/script-AssignAnalystToIncident.yml

```diff
@@ -0,0 +1,55 @@
+commonfields:
+  id: AssignAnalystToIncident
+  version: -1
+name: AssignAnalystToIncident
+system: true
+script: |-
+  from random import choice
+
+  rolesList = []
+  if demisto.get(demisto.args(), 'roles'):
+      rolesList = argToList(demisto.args()['roles']);
+
+  isRandom = True
+  if demisto.get(demisto.args(), 'isRandom'):
+      if demisto.args()['isRandom'].lower() == 'no':
+          isRandom = False
+
+  res = demisto.executeCommand('getUsers', { 'roles': ','.join(rolesList) });
+  usersList = list(map((lambda u: u['username']), res[0]['Contents']))
+
+  userToAssign = None
+  if isRandom:
+      userToAssign = choice(usersList)
+  else:
+      usersMap = dict.fromkeys(usersList, 0)
+      assignedIncidents = demisto.executeCommand("getIncidents", { 'status': '1' })[0]['Contents']['data']
```

Comment:
1. you will get at most 100 incidents with this implementation.
2. might be that all returned incidents will not be assigned.